### ISSUE TITLE
Make the new search dialog accessible

### ISSUE THREAD

**Issue by kittykat** [vector-im/element-web#22361]
The new search dialog needs to be accessible, especially for screen reader users.

Resources:
* https://github.com/matrix-org/element-web-rageshakes/issues/10871

**Comment by justjanne** [comment]
The user ticket gives us the following tasks:

- [X] The search input box is not visible to screenreaders when opening the new search  
  https://github.com/matrix-org/matrix-react-sdk/pull/7856
- [ ] The room list / space list can not be reached via a shortcut directly
- [ ] Rooms and Users should have a context menu in search results the same as if they’re shown in a room or member list

**Comment by niquewoodhouse** [comment]
> * [ ]  The room list / space list can not be reached via a shortcut directly

Maybe I am misunderstanding, but I don't understand why this is relevant specifically to the new search? The room list and space list are separate features which currently don't have shortcuts, do they?  But A user can use a keyboard shortcut cmd + f in chrome to find text in the UI for room list - or is that taken up by something else in element?

> * [ ]  Rooms and Users should have a context menu in search results the same as if they’re shown in a room or member list

As in, give them these icons/the options within them?
<img width="48" alt="image" src="https://user-images.githubusercontent.com/1587166/174119237-b2ca5640-8a18-4d9f-b73a-2841b87f8d5e.png">

**Comment by justjanne** [comment]
We'll move the discussion regarding the context menus and the requested shortcuts for the space/room list into separate issues
